### PR TITLE
Core/Auth: Cleanup mute system

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: sql/updates/auth/2015_09_23_00_auth.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE `account` DROP COLUMN `mutetime`, DROP COLUMN `mutereason`, DROP COLUMN `muteby`;
```
Comment (DDuarte) [inline sql/updates/auth/2015_09_23_00_auth.sql]: These changes need to be reflected in the `auth_database.sql` file (use the `mysqldump command and then reset not required changes, like the changing mysql version in the file).

Comment (mthsena) [inline sql/updates/auth/2015_09_23_00_auth.sql]: Done.

Comment (Aokromes) [inline sql/updates/auth/2015_09_23_00_auth.sql]: sql/updates/auth/2015_09_23_00_auth.sql

I sugest you to name filenames as the PR id you are ussing ie:

sql/updates/auth/2015_09_23_15584_auth.sql

File: src/server/database/Database/Implementation/LoginDatabase.cpp [hunk 2]
```diff
@@ -82,7 +80,7 @@ void LoginDatabaseConnection::DoPrepareStatements()
     PrepareStatement(LOGIN_GET_USERNAME_BY_ID, "SELECT username FROM account WHERE id = ?", CONNECTION_SYNCH);
     PrepareStatement(LOGIN_SEL_CHECK_PASSWORD, "SELECT 1 FROM account WHERE id = ? AND sha_pass_hash = ?", CONNECTION_SYNCH);
     PrepareStatement(LOGIN_SEL_CHECK_PASSWORD_BY_NAME, "SELECT 1 FROM account WHERE username = ? AND sha_pass_hash = ?", CONNECTION_SYNCH);
-    PrepareStatement(LOGIN_SEL_PINFO, "SELECT a.username, aa.gmlevel, a.email, a.reg_mail, a.last_ip, DATE_FORMAT(a.last_login, '%Y-%m-%d %T'), a.mutetime, a.mutereason, a.muteby, a.failed_logins, a.locked, a.OS FROM account a LEFT JOIN account_access aa ON (a.id = aa.id AND (aa.RealmID = ? OR aa.RealmID = -1)) WHERE a.id = ?", CONNECTION_SYNCH);
+    PrepareStatement(LOGIN_SEL_PINFO, "SELECT a.username, aa.gmlevel, a.email, a.reg_mail, a.last_ip, DATE_FORMAT(a.last_login, '%Y-%m-%d %T'), a.failed_logins, a.locked, a.OS FROM account a LEFT JOIN account_access aa ON (a.id = aa.id AND (aa.RealmID = ? OR aa.RealmID = -1)) WHERE a.id = ?", CONNECTION_SYNCH);
```
Comment (joschiwald) [inline src/server/database/Database/Implementation/LoginDatabase.cpp]: you forgot to adjust the place where this query is used?

File: src/server/database/Database/Implementation/LoginDatabase.cpp [hunk 3]
```diff
@@ -112,5 +110,7 @@ void LoginDatabaseConnection::DoPrepareStatements()
...
     PrepareStatement(LOGIN_DEL_RBAC_ACCOUNT_PERMISSION, "DELETE FROM rbac_account_permissions WHERE accountId = ? AND permissionId = ? AND (realmId = ? OR realmId = -1)", CONNECTION_ASYNC);
     PrepareStatement(LOGIN_INS_ACCOUNT_MUTE, "INSERT INTO account_muted VALUES (?, UNIX_TIMESTAMP(), ?, ?, ?)", CONNECTION_ASYNC);
-    PrepareStatement(LOGIN_SEL_ACCOUNT_MUTE_INFO, "SELECT mutedate, mutetime, mutereason, mutedby FROM account_muted WHERE guid = ? ORDER BY mutedate ASC", CONNECTION_SYNCH);
+    PrepareStatement(LOGIN_SEL_ACCOUNT_MUTE_INFO, "SELECT mutedate, mutetime, mutereason, mutedby FROM account_muted WHERE account = ? ORDER BY mutedate ASC", CONNECTION_SYNCH);
+    PrepareStatement(LOGIN_UPD_ACCOUNT_MUTE_TIME, "UPDATE account_muted SET mutetime = ? , mutereason = ? , muteby = ? WHERE account = ?", CONNECTION_ASYNC);
+    PrepareStatement(LOGIN_UPD_ACCOUNT_MUTE_TIME_LOGIN, "UPDATE account_muted SET mutetime = ? WHERE account = ?", CONNECTION_ASYNC);
```
Comment (joschiwald) [inline src/server/database/Database/Implementation/LoginDatabase.cpp]: these update queries won't work correctly, fk is on account and mutetime (table is like a history system)

**Comment by Nyeriah** [comment]
The account_muted is indeed aimed to be a history system, so, this PR looks like a regression to me. What exactly is this aimed at and what improvements can we expect, other than the information no longer being stored?

Could just as well revert the PR that implemented storing it, mutes worked before without the history table (if you check `account` you'll see that current mutetime and mutereason are still present there).

**Comment by mthsena** [comment]
Apparently I thought it was account_muted the system, and I did not really know what was just to save the history.

The table should have a name related to the proposed.

**Comment by tkrokli** [comment]
You mean something like `account_muted_history` or `account_mute_log` ?

**Comment by mthsena** [comment]
Yes, but I think that this PR still can be availed just  adding a column ```active``` like ban system.

**Pull request closed without merging**